includes/class-amp-theme-support.php

```diff
@@ -1422,7 +1422,31 @@ protected static function has_dependency( WP_Dependencies $dependencies, $curren
 	 * @return string Tag.
 	 */
 	public static function filter_admin_bar_style_loader_tag( $tag, $handle ) {
-		if ( self::has_dependency( wp_styles(), $handle, 'admin-bar' ) ) {
+		if ( 'dashicons' === $handle ) {
+			// Conditionally include Dashicons in dev mode only if was included because it is a dependency of admin-bar.
+			$needs_dev_mode = true;
+			foreach ( wp_styles()->queue as $queued_handle ) {
+				$is_dashicons_without_admin_bar = (
```

Comment:
The variable `$is_dashicons_without_admin_bar` doesn't fulfil a function here, you can just replace it with `if` and immediately set the dev mode:
```php
				if (
					// If a theme or plugin directly enqueued dashicons, then it is not added via admin-bar dependency and it is not part of dev mode.
					'dashicons' === $queued_handle
					||
					// If a stylesheet has dashicons as a dependency without also having admin-bar as a dependency, then no dev mode.
					(
						self::has_dependency( wp_styles(), $queued_handle, 'dashicons' )
						&&
						! self::has_dependency( wp_styles(), $queued_handle, 'admin-bar' )
					)
				) {
					$needs_dev_mode = false;
					break;
				}
```

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3187/commits/5de22ffafdbb65ea87b5aeb97635472447a6a194